imageprocessor/imageprocessor-core/src/main/java/pl/edu/uj/kimage/processingFlow/StepResultEvent.java

```diff
@@ -0,0 +1,49 @@
+package pl.edu.uj.kimage.processingFlow;
+
+import pl.edu.uj.kimage.eventbus.Event;
+
+public class StepResultEvent<T> implements Event {
+    private final int stepId;
+    private final T result;
+
+    public StepResultEvent(int stepId, T result) {
+        this.stepId = stepId;
+        this.result = result;
+    }
+
+    public int getStepId() {
+        return stepId;
+    }
+
+    public Class<?> resultClass()
+    {
+        return result.getClass();
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        StepResultEvent<?> that = (StepResultEvent<?>) o;
+
+        if (stepId != that.stepId) return false;
+        return result != null ? result.equals(that.result) : that.result == null;
```

Comment:
Did you think about that code could be wrong? When you compare `StepResultEvent` from two threads it could cause bugs.